frontends/PyCDE/src/pycde/instance.py

```diff
@@ -137,7 +135,7 @@ def _attach_attribute(self, attr_key: str, attr: ir.Attribute):
       global_ref = hw.GlobalRefOp(global_ref_symbol, path_attr)
 
     # Attach the attribute to the global ref.
-    global_ref.attributes["loc:" + attr_key[4:]] = attr
+    global_ref.attributes["loc:" + sub_path] = attr
```

Comment:
Can make a new 'msft.locations' attribute and have it be an array of `PhysLocationAttr`s? Or is now the time to move to a new `msft.locate @gref1 <physLocAttr>` (which is probably the more principled thing to do)?

Reply:
I'm not sure I follow what `msft.locations` is. Do you mean to use an ArrayAttr, or add an AttrDef that is basically just ArrayAttr? Or are you just talking about how to name the attribute when we add it to an op's attribute dictionary?

I don't really see much value from defining `msft.locate` as a separate op. Would it look like this?

```
hw.globalRef @gref1 [@top::@foo, @bar::@baz]
msft.locate @gref1 #msft.physloc<...>
```

I don't see much material gain versus the current approach (modulo naming of the attribute):

```
hw.globalRef @gref1 [@top::@foo, @bar::@baz] { msft.locate = #msft.physloc<...> }
```

IMO adding a discardable attribute with a name prefixed under the msft dialect is the cleanest way to go. The passes that care about them can use them, and when it comes time to strip global refs out after Tcl export, we can look for this attribute.

Reply:
After discussing in ODM, we will go with the first option, a separate `msft.locate` option.

Reply:
Thanks, this matches firrtl dialect's non-local annotation implementation approach.